### PR TITLE
Add cardano-node and cardano-cli to the nix-shell.

### PR DESCRIPTION
**Pull request by raduom**
Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -91,6 +91,12 @@ rec {
     inherit haskell webCommon webCommonPlutus;
   });
 
+  # Cardano
+  inherit (plutus.haskell.project.hsPkgs.cardano-cli.components.exes) cardano-cli;
```
Comment (michaelpj) [inline default.nix:95]: Can we put this in `nix/pkgs/default.nix`? That already has a bunch of executables pulled out from haskell packages defined there.

File: default.nix [hunk 2]
```diff
@@ -91,6 +91,12 @@ rec {
...
+  cardano-node = plutus.haskell.project.hsPkgs.cardano-node.components.exes.cardano-node // {
+    deployments = pkgs.cardano-node-deployments;
```
Comment (michaelpj) [inline default.nix:97]: This definitely needs a comment!

Comment (raduom) [reply default.nix:97]: This incantation was lifted from the `cardano-wallet` project, I don't know what it does yet :)

**Comment by michaelpj** [comment]
Okay, I think this should be good now.

**Pull request merged**